core/src/main/java/org/apache/iceberg/PartitionStatsHandler.java

```diff
@@ -336,17 +337,23 @@ private static PartitionMap<PartitionStats> computeStatsDiff(
         Sets.newHashSet(
             SnapshotUtil.ancestorIdsBetween(
                 toSnapshot.snapshotId(), fromSnapshot.snapshotId(), table::snapshot));
-    Predicate<ManifestFile> manifestFilePredicate =
-        manifestFile -> snapshotIdsRange.contains(manifestFile.snapshotId());
-    return computeStats(table, toSnapshot, manifestFilePredicate, true /* incremental */);
+    // DELETED manifest entries are not carried over to subsequent snapshots.
+    // So, for incremental computation, gather the manifests added by each snapshot
+    // instead of relying solely on those from the latest snapshot.
+    List<ManifestFile> manifests =
+        snapshotIdsRange.stream()
```

Comment:
nit: How about just call [SnapshotUtil::ancestorsBetween](https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java#L208) and iterate through the ancestors?

Reply:
Updated.